File: frag-nav/src/test/java/com/ncapdevi/fragnav/tabhistory/CurrentTabHistoryControllerTest.java
package com.ncapdevi.fragnav.tabhistory;

import com.ncapdevi.fragnav.FragNavPopController;
import com.ncapdevi.fragnav.FragNavTransactionOptions;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.ArgumentMatchers.isNull;
import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class CurrentTabHistoryControllerTest {
    @Mock
    private FragNavPopController mockFragNavPopController;

    @Test
    public void testPopDelegatedWhenPopCalled() {
        // Given
        CurrentTabHistoryController currentTabHistoryController = new CurrentTabHistoryController(
                mockFragNavPopController);

        // When
        currentTabHistoryController.popFragments(1, null);

        // Then
        verify(mockFragNavPopController, times(1)).tryPopFragments(eq(1), isNull(FragNavTransactionOptions.class));
    }
}

File: frag-nav/src/test/java/com/ncapdevi/fragnav/tabhistory/CurrentTabHistoryControllerTest.kt
package com.ncapdevi.fragnav.tabhistory

import com.ncapdevi.fragnav.FragNavPopController
import com.nhaarman.mockitokotlin2.isNull
import com.nhaarman.mockitokotlin2.times
import com.nhaarman.mockitokotlin2.verify
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.ArgumentMatchers.eq
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class CurrentTabHistoryControllerTest {
    @Mock
    private lateinit var mockFragNavPopController: FragNavPopController

    @Test
    fun testPopDelegatedWhenPopCalled() {
        // Given
        val currentTabHistoryController = CurrentTabHistoryController(mockFragNavPopController)

        // When
        currentTabHistoryController.popFragments(1, null)

        // Then
        verify(mockFragNavPopController, times(1)).tryPopFragments(eq(1), isNull())
    }
}